File: server/src/main/java/br/app/pregsrateio/common/erros/ErroResponse.java
package br.app.pregsrateio.common.erros;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;
import java.util.Map;
import java.util.Objects;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ErroResponse {

    private String codigo;
    private String mensagem;
    private List<Map<String, Object>> detalhes;

    // Constante de erro inesperado
    public static final ErroResponse INESPERADO = ErroResponse.builder()
            .codigo("ERRO_INESPERADO")
            .mensagem("Ocorreu um erro inesperado. Tente novamente mais tarde.")
            .build();
}

File: server-old/src/main/kotlin/br/dev/brunorsch/pregsrateio/common/erros/ErroResponse.kt
package br.dev.brunorsch.pregsrateio.common.erros

import io.micronaut.serde.annotation.Serdeable

@Serdeable
data class ErroResponse(
    val codigo: String,
    val mensagem: String,
    val detalhes: List<Map<String, Any>> = emptyList(),
) {
    companion object {
        val INESPERADO = ErroResponse(
            codigo = "INTERNAL_SERVER_ERROR",
            mensagem = "Ocorreu um erro inesperado. Tente novamente mais tarde.",
            detalhes = emptyList()
        )
    }
}